src/parse.rs

```diff
@@ -72,21 +72,21 @@ impl<'a> Reader<'a> {
 }
 
 fn is_space(ch: u8) -> bool {
-    return ch == b' ' || ch == b'\t' || ch == b'\n';
+    ch == b' ' || ch == b'\t' || ch == b'\n'
 }
 
 fn is_separator(ch: u8) -> bool {
-    return is_space(ch)
+    is_space(ch)
         || ch == b'('
         || ch == b')'
         || ch == b'{'
         || ch == b'}'
         || ch == b'['
         || ch == b']'
-        || ch == b'.';
+        || ch == b'.'
 }
 
-fn skip_space<'a>(r: &mut Reader<'a>) {
+fn skip_space(r: &mut Reader<'_>) {
```

Comment:
This might be my ignorance, but are these equivalent? Isn't it more flexible to take the lifetime as a parameter?

Reply:
These are exactly equivalent. If you don't return any reference, you can avoid using lifetimes. See https://doc.rust-lang.org/nomicon/lifetime-elision.html for more infos.